tesseract_core/sdk/templates/pytorch/tesseract_api.py

```diff
@@ -78,23 +78,28 @@ def jacobian(
     # flatten the dict such that they can be accessed by paths and obtain the values
     pos_inputs = flatten_with_paths(tensor_inputs, jac_inputs).values()
 
-    # create a positional function that accepts a list of values and returns a set of tuples
+    # create a positional function that accepts a flat sequence of values and returns a tuple
     filtered_pos_eval = filter_func(
         evaluate,
         tensor_inputs,
         jac_outputs,
         input_paths=jac_inputs,
-        output_to_tuple=True,
     )
 
+    def filtered_pos_eval_flat(*args):
+        res = filtered_pos_eval(*args)
+        return tuple(res[k] for k in jac_outputs)
```

Comment:
I don't think this would work if you had left `jac_outputs` out as I think you meant to. If you didn't mean too then this would suffice (although the function would already be flat so the naming is misleading).
```suggestion
    filtered_pos_eval_flat = lambda *args: tuple(filtered_pos_eval(*args).values())
```
The Python3 ability to use kwargs as positional arguments seems to often lead to confusion (e.g. lack of clarity whether a kwarg is used or was meant to be used), do you think there it is worth looking at trying to '/' and '*' notation more to enforce this?

Reply:
I don't get it. The code is exactly as it's supposed to be.

Reply:
Then why leave `output_paths` as optional if it's always called? Quick call?

Reply:
(Yes ideally we'd use `kwargs` for those but that would be a breaking change for the JAX template so I went for backwards compatibility instead.)

Reply:
Ok, I get the argument about why to make it optional in general. Do you agree though that as you are already providing `jac_outputs` that `filtered_pos_eval` _already_ returns a flat filtered output, and the purpose of the lines highlighted in this conversation is purely to convert to tuple and is thus overkill as is?

Reply:
No, converting to tuple is the purpose of the function. I could give you a long explanation of why I like this a few % better than the one-liner with the lambda, but let's please converse our energy to argue the details that actually matter.

Reply:
Ok, I will leave this up to you. 🙂 Last $0.02 is that the function name is probably the most important thing for avoiding confusion (i.e. `filtered_pos_eval_tuple` would be better than `filtered_pos_eval_flat`).

Reply:
That seems to imply that the assigned object is a tuple, when it's really a function. I agree the naming isn't perfect here, and thanks for your input and concern for UX. Let's view this as an exercise of leaving some low-impact optimizations on the table in the interest of efficiency – this review has dragged on long enough already.